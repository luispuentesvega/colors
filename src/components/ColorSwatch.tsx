import { useState, useEffect } from 'react';

import Swatch from '@uiw/react-color-swatch';

import { ColorService } from '../utils/factory/ColorService';

const ColorSwatch: React.FC = () => {
  const [colors, setColors] = useState([]);

  useEffect(() => {
    fetchColors();
  }, []);

  const fetchColors = async (): Promise<void> => {
    try {
      const response = await fetch(`api/color`);
      const result = await response.json();
      const mappedColors = result.map((color: any) => new ColorService().getColorAsString(color.type, { ...color }));
      setColors(mappedColors);
    } catch (e) {
      console.log('ERROR', e);
    }
  };

  const handleChangeColors = () => {
    fetchColors();
  };

  return (
    <div className="bg-orange-300 px-8 py-8 rounded-lg">
      <h1 className="text-3xl font-bold text-violet-500 mb-10">Color Swatch</h1>
      <Swatch colors={colors}
        style={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        }} />
      <button onClick={handleChangeColors} type="button" className="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-hidden focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 w-full mt-10">Refresh</button>
    </div>
  );
};

export default ColorSwatch;
