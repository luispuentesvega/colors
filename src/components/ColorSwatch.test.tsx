import { render, screen } from '@testing-library/react';
import ColorSwatch from './ColorSwatch';
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve([{ type: 'rgb' }]),
  }),
) as jest.Mock;

describe('ColorSwatch', () => {
  it('should render Color Swatch title', () => {
    const { container } = render(<ColorSwatch />);

    expect(container).toMatchSnapshot();
  });
});
