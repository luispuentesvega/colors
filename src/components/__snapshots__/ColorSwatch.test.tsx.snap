// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwatch should match snaptshot 1`] = `
<div
  className="bg-orange-300 px-8 py-8 rounded-lg"
>
  <h1
    className="text-3xl font-bold text-violet-500 mb-10"
  >
    Color Swatch
  </h1>
  <div
    className="w-color-swatch"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "position": "relative",
      }
    }
  />
  <button
    className="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-hidden focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 w-full mt-10"
    onClick={[Function]}
    type="button"
  >
    Refresh
  </button>
</div>
`;

exports[`ColorSwatch should render Color Swatch title 1`] = `
<div>
  <div
    class="bg-orange-300 px-8 py-8 rounded-lg"
  >
    <h1
      class="text-3xl font-bold text-violet-500 mb-10"
    >
      Color Swatch
    </h1>
    <div
      class="w-color-swatch"
      style="display: flex; flex-wrap: wrap; position: relative; align-items: center; justify-content: center;"
    />
    <button
      class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-hidden focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 w-full mt-10"
      type="button"
    >
      Refresh
    </button>
  </div>
</div>
`;
